src/interface/ui/blockchainaccountstatus/BlockchainAccountStatus.cpp

```diff
@@ -507,5 +516,16 @@ auto BlockchainAccountStatus::subchain_display_name(
     return out;
 }
 
-BlockchainAccountStatus::~BlockchainAccountStatus() = default;
+BlockchainAccountStatus::~BlockchainAccountStatus()
+{
+    try {
+        if (check_running_mark_stopped()) {
+            close_pipeline();
+            // TODO MT-34 investigate what other actions might be needed
+            mark_shut_down();
+        }
+    } catch (const std::exception&) {
+        // TODO MT-34 diagnostics
```

Comment:
I suggest to print what went wrong before MT-34 will be implemented

Reply:
More is needed, hence the TODO and ticket, but I have added a minimal diagnostic line especially for you :-)